From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sam Maier <smaier@chromium.org>
Date: Tue, 4 Jul 2023 18:41:35 +0000
Subject: [PATCH] Android U: rolling DEPS entries to start using Android U by
 default

Bug: 1445802, 1417656
Change-Id: I527e884638ec7e4f6b5572cfea44ef1c8425448e
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4621418
Auto-Submit: Sam Maier <smaier@chromium.org>
Code-Coverage: Findit <findit-for-me@appspot.gserviceaccount.com>
Commit-Queue: Sam Maier <smaier@chromium.org>
Reviewed-by: Andrew Grieve <agrieve@chromium.org>
Commit-Queue: Andrew Grieve <agrieve@chromium.org>
Owners-Override: Andrew Grieve <agrieve@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1165703}
---
 DEPS                                          | 12 +++++-----
 ...em_webview_bundle.AndroidManifest.expected |  9 +++++---
 ...me_webview_bundle.AndroidManifest.expected |  9 +++++---
 .../nonembedded/java/AndroidManifest.xml      |  9 --------
 android_webview/system_webview_apk_tmpl.gni   |  4 ----
 .../java/src/org/chromium/base/BuildInfo.java | 20 ++++++++---------
 build/android/pylib/constants/__init__.py     |  2 +-
 build/config/android/config.gni               | 10 ++++-----
 build/config/android/sdk.gni                  |  4 ++--
 chrome/android/expectations/lint-baseline.xml | 22 +++++++++----------
 ...blic_bundle__base.AndroidManifest.expected |  8 ++++---
 ...ev2_authenticator.AndroidManifest.expected |  6 ++---
 ...ic_bundle__chrome.AndroidManifest.expected |  6 ++---
 ...ic_bundle__dev_ui.AndroidManifest.expected |  6 ++---
 ...e__stack_unwinder.AndroidManifest.expected |  6 ++---
 ...undle__test_dummy.AndroidManifest.expected |  6 ++---
 ...public_bundle__vr.AndroidManifest.expected |  6 ++---
 ..._bundle__weblayer.AndroidManifest.expected |  6 ++---
 ...rome_bundle__base.AndroidManifest.expected |  6 ++---
 ...hrome_library_apk.AndroidManifest.expected |  6 ++---
 .../android/apk/AndroidManifest.xml.jinja2    | 10 ++++++++-
 third_party/android_sdk/README.chromium       |  8 +++----
 .../android_system_sdk/README.chromium        |  2 +-
 23 files changed, 93 insertions(+), 90 deletions(-)

diff --git a/DEPS b/DEPS
index 26a69b796e171..b6c787958b09c 100644
--- a/DEPS
+++ b/DEPS
@@ -403,7 +403,7 @@ vars = {
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling android_sdk_build-tools_version
   # and whatever else without interference from each other.
-  'android_sdk_build-tools_version': '-VRKr36Uw8L_iFqqo9nevIBgNMggND5iWxjidyjnCgsC',
+  'android_sdk_build-tools_version': 'YK9Rzw3fDzMHVzatNN6VlyoD_81amLZpN1AbmkdOd6AC',
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling android_sdk_emulator_version
   # and whatever else without interference from each other.
@@ -419,7 +419,7 @@ vars = {
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling android_sdk_platform-tools_version
   # and whatever else without interference from each other.
-  'android_sdk_platform-tools_version': 'RSI3iwryh7URLGRgJHsCvUxj092woTPnKt4pwFcJ6L8C',
+  'android_sdk_platform-tools_version': 'HWVsGs2HCKgSVv41FsOcsfJbNcB0UFiNrF6Tc4yRArYC',
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling android_sdk_platforms_version
   # and whatever else without interference from each other.
@@ -427,7 +427,7 @@ vars = {
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling android_sdk_platforms_version
   # and whatever else without interference from each other.
-  'android_sdk_platforms_version': 'eo5KvW6UVor92LwZai8Zulc624BQZoCu-yn7wa1z_YcC',
+  'android_sdk_platforms_version': 'u-bhWbTME6u-DjypTgr3ZikCyeAeU6txkR9ET6Uudc8C',
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling android_sdk_sources_version
   # and whatever else without interference from each other.
@@ -1021,7 +1021,7 @@ deps = {
       'packages': [
           {
               'package': 'chromium/third_party/android_system_sdk/public',
-              'version': 'RGY8Vyf8jjszRIJRFxZj7beXSUEHTQM90MtYejUvdMgC',
+              'version': '4QeolYaSKWBtVTgzJU4tHUfzA9OJTDM8YUcD426IctwC',
           },
       ],
       'condition': 'checkout_android',
@@ -1086,7 +1086,7 @@ deps = {
   'src/third_party/android_sdk/public': {
       'packages': [
           {
-              'package': 'chromium/third_party/android_sdk/public/build-tools/33.0.0',
+              'package': 'chromium/third_party/android_sdk/public/build-tools/34.0.0',
               'version': Var('android_sdk_build-tools_version'),
           },
           {
@@ -1102,7 +1102,7 @@ deps = {
               'version': Var('android_sdk_platform-tools_version'),
           },
           {
-              'package': 'chromium/third_party/android_sdk/public/platforms/android-33',
+              'package': 'chromium/third_party/android_sdk/public/platforms/android-34',
               'version': Var('android_sdk_platforms_version'),
           },
           {
diff --git a/android_webview/expectations/system_webview_bundle.AndroidManifest.expected b/android_webview/expectations/system_webview_bundle.AndroidManifest.expected
index 0073c63600484..7eaff182b4fde 100644
--- a/android_webview/expectations/system_webview_bundle.AndroidManifest.expected
+++ b/android_webview/expectations/system_webview_bundle.AndroidManifest.expected
@@ -2,8 +2,8 @@
 <manifest
     xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.android.webview"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:isolatedSplits="true"
     android:versionCode="OFFSET=0"
     android:versionName="#.#.#.#">
@@ -11,10 +11,11 @@
   <uses-feature android:name="android.software.leanback" android:required="false"/>
   <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+  <uses-permission android:name="android.permission.FOREGROUND_SERVICE_SPECIAL_USE"/>
   <uses-permission android:name="android.permission.INTERNET"/>
   <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
   <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <application
       android:name="org.chromium.android_webview.nonembedded.WebViewApkApplication"
       android:extractNativeLibs="false"
@@ -139,7 +140,9 @@
     <service  # DIFF-ANCHOR: adce9ea1
         android:name="org.chromium.android_webview.services.DeveloperUiService"
         android:exported="true"
+        android:foregroundServiceType="specialUse"
         android:process=":webview_service">
+      <property android:name="android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE" android:value="debugging"/>
     </service>  # DIFF-ANCHOR: adce9ea1
     <service  # DIFF-ANCHOR: eecf2fee
         android:name="org.chromium.android_webview.services.MetricsBridgeService"
diff --git a/android_webview/expectations/trichrome_webview_bundle.AndroidManifest.expected b/android_webview/expectations/trichrome_webview_bundle.AndroidManifest.expected
index 0ec064a7a3942..abac15d46ef50 100644
--- a/android_webview/expectations/trichrome_webview_bundle.AndroidManifest.expected
+++ b/android_webview/expectations/trichrome_webview_bundle.AndroidManifest.expected
@@ -2,8 +2,8 @@
 <manifest
     xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.android.webview"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:isolatedSplits="true"
     android:versionCode="OFFSET=30"
     android:versionName="#.#.#.#">
@@ -11,10 +11,11 @@
   <uses-feature android:name="android.software.leanback" android:required="false"/>
   <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+  <uses-permission android:name="android.permission.FOREGROUND_SERVICE_SPECIAL_USE"/>
   <uses-permission android:name="android.permission.INTERNET"/>
   <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
   <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
-  <uses-sdk android:minSdkVersion="29" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="29" android:targetSdkVersion="34"/>
   <application
       android:name="org.chromium.android_webview.nonembedded.WebViewApkApplication"
       android:extractNativeLibs="false"
@@ -139,7 +140,9 @@
     <service  # DIFF-ANCHOR: adce9ea1
         android:name="org.chromium.android_webview.services.DeveloperUiService"
         android:exported="true"
+        android:foregroundServiceType="specialUse"
         android:process=":webview_service">
+      <property android:name="android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE" android:value="debugging"/>
     </service>  # DIFF-ANCHOR: adce9ea1
     <service  # DIFF-ANCHOR: eecf2fee
         android:name="org.chromium.android_webview.services.MetricsBridgeService"
diff --git a/android_webview/nonembedded/java/AndroidManifest.xml b/android_webview/nonembedded/java/AndroidManifest.xml
index c9c2a4de6f872..8fdea244f2719 100644
--- a/android_webview/nonembedded/java/AndroidManifest.xml
+++ b/android_webview/nonembedded/java/AndroidManifest.xml
@@ -30,10 +30,7 @@ by a child template that "extends" this file.
     <!-- Required by WebView Developer UI only -->
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
 
-    {# TODO(wbjacksonjr crbug.com/1417656): Remove once we build against U #}
-    {% if android_sdk_release is defined and android_sdk_release == 'u' %}
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE_SPECIAL_USE"/>
-    {% endif %}
     <!-- Required to run the foreground service-->
     <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
 
@@ -167,20 +164,14 @@ by a child template that "extends" this file.
                          android:exported="true"
                          android:process=":webview_service" />  {# Explicit process required for monochrome compatibility. #}
                 <service android:name="org.chromium.android_webview.services.DeveloperUiService"
-                         {# TODO(wbjacksonjr crbug.com/1417656): Remove once we build against U#}
-                         {% if android_sdk_release is defined and android_sdk_release == 'u' %}
                          android:foregroundServiceType="specialUse"
-                         {% endif %}
                          android:exported="true"
                          android:process=":webview_service"  {# Explicit process required for monochrome compatibility. #}
                          tools:ignore="ExportedService" >
-                         {# TODO(wbjacksonjr crbug.com/1417656): Remove once we build against U#}
-                         {% if android_sdk_release is defined and android_sdk_release == 'u' %}
                           <property
                             android:name="android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE"
                             android:value="debugging"
                           />
-                         {% endif %}
                 </service>
                 <service android:name="org.chromium.android_webview.services.SafeModeService"
                          android:exported="true"
diff --git a/android_webview/system_webview_apk_tmpl.gni b/android_webview/system_webview_apk_tmpl.gni
index 3942af4ada251..e3524c7eee635 100644
--- a/android_webview/system_webview_apk_tmpl.gni
+++ b/android_webview/system_webview_apk_tmpl.gni
@@ -56,15 +56,11 @@ template("system_webview_apk_or_module_tmpl") {
   _android_manifest = "$target_gen_dir/$target_name/AndroidManifest.xml"
   _android_manifest_target_name = "${target_name}__android_manifest"
   jinja_template(_android_manifest_target_name) {
-    if (!defined(_android_sdk_release)) {
-      _android_sdk_release = android_sdk_release
-    }
     input = "//android_webview/nonembedded/java/AndroidManifest.xml"
     output = _android_manifest
     _force_32_bit = _include_32_bit_webview && _include_64_bit_webview &&
                     (!_is_trichrome || !_is_64_bit_browser)
     variables = [
-      "android_sdk_release=$_android_sdk_release",
       "force_32_bit=$_force_32_bit",
       "manifest_package=$_manifest_package",
     ]
diff --git a/base/android/java/src/org/chromium/base/BuildInfo.java b/base/android/java/src/org/chromium/base/BuildInfo.java
index d7907d41f9e56..faadb9d93aa9e 100644
--- a/base/android/java/src/org/chromium/base/BuildInfo.java
+++ b/base/android/java/src/org/chromium/base/BuildInfo.java
@@ -285,20 +285,20 @@ public class BuildInfo {
      * This must be manually maintained as the SDK goes through finalization!
      * Avoid depending on this if possible; this is only intended for WebView.
      */
-    @OptIn(markerClass = androidx.core.os.BuildCompat.PrereleaseSdkCheck.class)
     public static boolean targetsAtLeastU() {
         int target = ContextUtils.getApplicationContext().getApplicationInfo().targetSdkVersion;
 
-        // Logic for after API finalization but before public SDK release has to
-        // just hardcode the appropriate SDK integer. This will include Android
-        // builds with the finalized SDK, and also pre-API-finalization builds
-        // (because CUR_DEVELOPMENT == 10000).
-        return target >= 34;
+        // Logic for pre-API-finalization:
+        // return BuildCompat.isAtLeastU() && target == Build.VERSION_CODES.CUR_DEVELOPMENT;
+
+        // Logic for after API finalization but before public SDK release has to just hardcode the
+        // appropriate SDK integer. This will include Android builds with the finalized SDK, and
+        // also pre-API-finalization builds (because CUR_DEVELOPMENT == 10000).
+        // return target >= 34;
 
-        // Once the public SDK is upstreamed we can use the defined constant,
-        // deprecate this, then eventually inline this at all callsites and
-        // remove it.
-        // return target >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE;
+        // Now that the public SDK is upstreamed we can use the defined constant. All users of this
+        // should now just inline this check themselves.
+        return target >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE;
     }
 
     public static void setFirebaseAppId(String id) {
diff --git a/build/android/pylib/constants/__init__.py b/build/android/pylib/constants/__init__.py
index cf57d9fe91bb0..2e978e0a1050f 100644
--- a/build/android/pylib/constants/__init__.py
+++ b/build/android/pylib/constants/__init__.py
@@ -114,7 +114,7 @@ DEVICE_PERF_OUTPUT_DIR = (
 
 SCREENSHOTS_DIR = os.path.join(DIR_SOURCE_ROOT, 'out_screenshots')
 
-ANDROID_SDK_BUILD_TOOLS_VERSION = '33.0.0'
+ANDROID_SDK_BUILD_TOOLS_VERSION = '34.0.0'
 ANDROID_SDK_ROOT = os.path.join(DIR_SOURCE_ROOT, 'third_party', 'android_sdk',
                                 'public')
 ANDROID_SDK_TOOLS = os.path.join(ANDROID_SDK_ROOT,
diff --git a/build/config/android/config.gni b/build/config/android/config.gni
index e4b8c2b0abe21..a327ee3da42a3 100644
--- a/build/config/android/config.gni
+++ b/build/config/android/config.gni
@@ -97,12 +97,12 @@ if (is_android || is_chromeos) {
 
   public_android_sdk_root = "//third_party/android_sdk/public"
   public_android_sdk_build_tools =
-      "${public_android_sdk_root}/build-tools/33.0.0"
-  public_android_sdk_version = "33"
-  if (android_sdk_release == "t") {
+      "${public_android_sdk_root}/build-tools/34.0.0"
+  public_android_sdk_version = "34"
+  if (android_sdk_release == "u") {
     default_android_sdk_root = public_android_sdk_root
     default_android_sdk_version = public_android_sdk_version
-    default_android_sdk_build_tools_version = "33.0.0"
+    default_android_sdk_build_tools_version = "34.0.0"
     public_android_sdk = true
   }
 
@@ -122,7 +122,7 @@ if (is_android || is_chromeos) {
     # Purposefully repeated so that downstream can change
     # default_android_sdk_root without changing lint version.
     default_lint_android_sdk_root = public_android_sdk_root
-    default_lint_android_sdk_version = 33
+    default_lint_android_sdk_version = 34
   }
 
   if (!defined(default_extras_android_sdk_root)) {
diff --git a/build/config/android/sdk.gni b/build/config/android/sdk.gni
index fb39315c46406..45e3404924bd6 100644
--- a/build/config/android/sdk.gni
+++ b/build/config/android/sdk.gni
@@ -4,10 +4,10 @@
 
 # The default SDK release used by public builds. Value may differ in
 # internal builds.
-default_android_sdk_release = "t"
+default_android_sdk_release = "u"
 
 # SDK releases against which public builds are supported.
 public_sdk_releases = [
-  "t",
   "tprivacysandbox",
+  "u",
 ]
diff --git a/chrome/android/expectations/lint-baseline.xml b/chrome/android/expectations/lint-baseline.xml
index 651074c980daf..f1bcbe45fc786 100644
--- a/chrome/android/expectations/lint-baseline.xml
+++ b/chrome/android/expectations/lint-baseline.xml
@@ -5165,6 +5165,17 @@
             column="94"/>
     </issue>
 
+    <issue
+        id="ForegroundServicePermission"
+        message="foregroundServiceType:specialUse requires permission:[android.permission.FOREGROUND_SERVICE_SPECIAL_USE]"
+        errorLine1="    &lt;service android:name=&quot;org.chromium.android_webview.services.DeveloperUiService&quot; android:foregroundServiceType=&quot;specialUse&quot; android:exported=&quot;true&quot; android:process=&quot;:webview_service&quot; tools:ignore=&quot;ExportedService&quot;>"
+        errorLine2="    ^">
+        <location
+            file="gen/chrome/android/monochrome_public_bundle__lint/gen/chrome/android/monochrome_public_bundle__base_bundle_module/AndroidManifest.xml"
+            line="605"
+            column="5"/>
+    </issue>
+
     <issue
         id="UseSwitchCompatOrMaterialCode"
         message="Use `SwitchCompat` from AppCompat or `SwitchMaterial` from Material library"
@@ -7145,17 +7156,6 @@
             column="12"/>
     </issue>
 
-    <issue
-        id="UnusedResources"
-        message="The resource `R.macro.default_bg_color_elev_0` appears to be unused"
-        errorLine1="    &lt;macro name=&quot;default_bg_color_elev_0&quot;>?attr/colorSurface&lt;/macro>"
-        errorLine2="           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~">
-        <location
-            file="../../components/browser_ui/styles/android/java/res/values/semantic_colors_dynamic.xml"
-            line="9"
-            column="12"/>
-    </issue>
-
     <issue
         id="UsableSpace"
         message="Consider also using `StorageManager#getAllocatableBytes` and `allocateBytes` which will consider clearable cached data"
diff --git a/chrome/android/expectations/monochrome_public_bundle__base.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle__base.AndroidManifest.expected
index d06bf35ca9c88..3e0780ad1c041 100644
--- a/chrome/android/expectations/monochrome_public_bundle__base.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle__base.AndroidManifest.expected
@@ -2,8 +2,8 @@
 <manifest
     xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:isolatedSplits="true"
     android:versionCode="OFFSET=20"
     android:versionName="#.#.#.#">
@@ -75,7 +75,7 @@
   <uses-permission-sdk-23 android:name="android.permission.REQUEST_INSTALL_PACKAGES"/>
   <uses-permission-sdk-23 android:name="android.permission.USE_BIOMETRIC"/>
   <uses-permission-sdk-23 android:name="android.permission.USE_FINGERPRINT"/>
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <application
       android:name="org.chromium.chrome.browser.base.SplitMonochromeApplication"
       android:allowAudioPlaybackCapture="false"
@@ -306,7 +306,9 @@
     <service  # DIFF-ANCHOR: adce9ea1
         android:name="org.chromium.android_webview.services.DeveloperUiService"
         android:exported="true"
+        android:foregroundServiceType="specialUse"
         android:process=":webview_service">
+      <property android:name="android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE" android:value="debugging"/>
     </service>  # DIFF-ANCHOR: adce9ea1
     <service  # DIFF-ANCHOR: eecf2fee
         android:name="org.chromium.android_webview.services.MetricsBridgeService"
diff --git a/chrome/android/expectations/monochrome_public_bundle__cablev2_authenticator.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle__cablev2_authenticator.AndroidManifest.expected
index c2fb71a41b9f7..baf76373b8bbb 100644
--- a/chrome/android/expectations/monochrome_public_bundle__cablev2_authenticator.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle__cablev2_authenticator.AndroidManifest.expected
@@ -4,14 +4,14 @@
     xmlns:dist="http://schemas.android.com/apk/distribution"
     featureSplit="cablev2_authenticator"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:versionCode="OFFSET=20"
     android:versionName="#.#.#.#">
   <dist:module dist:onDemand="false" dist:title="@string/cablev2_authenticator_module_title">  # DIFF-ANCHOR: 4362cefb
     <dist:fusing dist:include="true"/>
   </dist:module>  # DIFF-ANCHOR: 4362cefb
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <uses-split android:name="chrome"/>
   <application/>
 </manifest>
diff --git a/chrome/android/expectations/monochrome_public_bundle__chrome.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle__chrome.AndroidManifest.expected
index c46158a95ae2f..0fc93940aa514 100644
--- a/chrome/android/expectations/monochrome_public_bundle__chrome.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle__chrome.AndroidManifest.expected
@@ -4,14 +4,14 @@
     xmlns:dist="http://schemas.android.com/apk/distribution"
     featureSplit="chrome"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:versionCode="OFFSET=20"
     android:versionName="#.#.#.#">
   <dist:module dist:onDemand="false">  # DIFF-ANCHOR: 2805007f
     <dist:fusing dist:include="true"/>
   </dist:module>  # DIFF-ANCHOR: 2805007f
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <application>
     <activity  # DIFF-ANCHOR: 28dc9019
         android:name="com.google.ar.core.InstallActivity"
diff --git a/chrome/android/expectations/monochrome_public_bundle__dev_ui.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle__dev_ui.AndroidManifest.expected
index fa6d89b5b467f..6a3a66607c92c 100644
--- a/chrome/android/expectations/monochrome_public_bundle__dev_ui.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle__dev_ui.AndroidManifest.expected
@@ -4,13 +4,13 @@
     xmlns:dist="http://schemas.android.com/apk/distribution"
     featureSplit="dev_ui"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:versionCode="OFFSET=20"
     android:versionName="#.#.#.#">
   <dist:module dist:onDemand="true" dist:title="@string/dev_ui_module_title">  # DIFF-ANCHOR: 671a5c9f
     <dist:fusing dist:include="true"/>
   </dist:module>  # DIFF-ANCHOR: 671a5c9f
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <application/>
 </manifest>
diff --git a/chrome/android/expectations/monochrome_public_bundle__stack_unwinder.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle__stack_unwinder.AndroidManifest.expected
index 6c340a133a4a0..eed9ff056ccd1 100644
--- a/chrome/android/expectations/monochrome_public_bundle__stack_unwinder.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle__stack_unwinder.AndroidManifest.expected
@@ -4,13 +4,13 @@
     xmlns:dist="http://schemas.android.com/apk/distribution"
     featureSplit="stack_unwinder"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:versionCode="OFFSET=20"
     android:versionName="#.#.#.#">
   <dist:module dist:onDemand="true" dist:title="@string/stack_unwinder_module_title">  # DIFF-ANCHOR: 2d9e963c
     <dist:fusing dist:include="true"/>
   </dist:module>  # DIFF-ANCHOR: 2d9e963c
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <application/>
 </manifest>
diff --git a/chrome/android/expectations/monochrome_public_bundle__test_dummy.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle__test_dummy.AndroidManifest.expected
index e5207239b7e21..0f17055a8043f 100644
--- a/chrome/android/expectations/monochrome_public_bundle__test_dummy.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle__test_dummy.AndroidManifest.expected
@@ -4,13 +4,13 @@
     xmlns:dist="http://schemas.android.com/apk/distribution"
     featureSplit="test_dummy"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:versionCode="OFFSET=20"
     android:versionName="#.#.#.#">
   <dist:module dist:onDemand="true" dist:title="@string/test_dummy_module_title">  # DIFF-ANCHOR: d029b25b
     <dist:fusing dist:include="true"/>
   </dist:module>  # DIFF-ANCHOR: d029b25b
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <application/>
 </manifest>
diff --git a/chrome/android/expectations/monochrome_public_bundle__vr.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle__vr.AndroidManifest.expected
index 0c46cd2a3d7bd..7fcc9ad47424c 100644
--- a/chrome/android/expectations/monochrome_public_bundle__vr.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle__vr.AndroidManifest.expected
@@ -4,8 +4,8 @@
     xmlns:dist="http://schemas.android.com/apk/distribution"
     featureSplit="vr"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:versionCode="OFFSET=20"
     android:versionName="#.#.#.#">
   <dist:module dist:instant="false" dist:title="@string/vr_module_title">  # DIFF-ANCHOR: aac219ae
@@ -19,7 +19,7 @@
     </dist:delivery>  # DIFF-ANCHOR: ea157ece
     <dist:fusing dist:include="true"/>
   </dist:module>  # DIFF-ANCHOR: aac219ae
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <uses-split android:name="chrome"/>
   <application/>
 </manifest>
diff --git a/chrome/android/expectations/monochrome_public_bundle__weblayer.AndroidManifest.expected b/chrome/android/expectations/monochrome_public_bundle__weblayer.AndroidManifest.expected
index 2c69b5466d7fa..19b5e3d745d02 100644
--- a/chrome/android/expectations/monochrome_public_bundle__weblayer.AndroidManifest.expected
+++ b/chrome/android/expectations/monochrome_public_bundle__weblayer.AndroidManifest.expected
@@ -4,14 +4,14 @@
     xmlns:dist="http://schemas.android.com/apk/distribution"
     featureSplit="weblayer"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:versionCode="OFFSET=20"
     android:versionName="#.#.#.#">
   <dist:module dist:onDemand="false">  # DIFF-ANCHOR: 2805007f
     <dist:fusing dist:include="true"/>
   </dist:module>  # DIFF-ANCHOR: 2805007f
-  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34"/>
   <uses-split android:name="chrome"/>
   <application/>
 </manifest>
diff --git a/chrome/android/expectations/trichrome_chrome_bundle__base.AndroidManifest.expected b/chrome/android/expectations/trichrome_chrome_bundle__base.AndroidManifest.expected
index 0d46f9b545630..9c1acfaa35b40 100644
--- a/chrome/android/expectations/trichrome_chrome_bundle__base.AndroidManifest.expected
+++ b/chrome/android/expectations/trichrome_chrome_bundle__base.AndroidManifest.expected
@@ -2,8 +2,8 @@
 <manifest
     xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.chromium.chrome.stable"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:isolatedSplits="true"
     android:versionCode="OFFSET=30"
     android:versionName="#.#.#.#">
@@ -75,7 +75,7 @@
   <uses-permission-sdk-23 android:name="android.permission.REQUEST_INSTALL_PACKAGES"/>
   <uses-permission-sdk-23 android:name="android.permission.USE_BIOMETRIC"/>
   <uses-permission-sdk-23 android:name="android.permission.USE_FINGERPRINT"/>
-  <uses-sdk android:minSdkVersion="29" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="29" android:targetSdkVersion="34"/>
   <application
       android:name="org.chromium.chrome.browser.base.SplitChromeApplication"
       android:allowAudioPlaybackCapture="false"
diff --git a/chrome/android/expectations/trichrome_library_apk.AndroidManifest.expected b/chrome/android/expectations/trichrome_library_apk.AndroidManifest.expected
index 0dace298d0c0b..40400f86d6897 100644
--- a/chrome/android/expectations/trichrome_library_apk.AndroidManifest.expected
+++ b/chrome/android/expectations/trichrome_library_apk.AndroidManifest.expected
@@ -2,14 +2,14 @@
 <manifest
     xmlns:android="http://schemas.android.com/apk/res/android"
     package="app.vanadium.trichromelibrary"
-    platformBuildVersionCode="33"
-    platformBuildVersionName="13"
+    platformBuildVersionCode="34"
+    platformBuildVersionName="14"
     android:versionCode="OFFSET=30"
     android:versionName="#.#.#.#">
   <uses-feature android:glEsVersion="0x00020000"/>
   <uses-feature android:name="android.hardware.touchscreen" android:required="false"/>
   <uses-feature android:name="android.software.leanback" android:required="false"/>
-  <uses-sdk android:minSdkVersion="29" android:targetSdkVersion="33"/>
+  <uses-sdk android:minSdkVersion="29" android:targetSdkVersion="34"/>
   <application
       android:extractNativeLibs="false"
       android:hasCode="false"
diff --git a/chromecast/browser/android/apk/AndroidManifest.xml.jinja2 b/chromecast/browser/android/apk/AndroidManifest.xml.jinja2
index c61278283a45d..a6fde7bb119a5 100644
--- a/chromecast/browser/android/apk/AndroidManifest.xml.jinja2
+++ b/chromecast/browser/android/apk/AndroidManifest.xml.jinja2
@@ -13,6 +13,7 @@
 
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_SPECIAL_USE"/>
     <uses-permission android:name="android.permission.INTERNET"/>
     <uses-permission android:name="android.permission.START_ACTIVITIES_FROM_BACKGROUND"
                      tools:ignore="ProtectedPermissions" />
@@ -28,13 +29,20 @@
                  android:hardwareAccelerated="true"
                  android:label="Chromecast Web Browser"
                  android:exported="true"
+                 android:foregroundServiceType="specialUse"
                  tools:ignore="ExportedService">
+             <property android:name="android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE"
+                       android:value="cast" />
             <intent-filter>
                 <action android:name="com.google.cast.action.START_CAST_BROWSER" />
                 <category android:name="android.intent.category.DEFAULT" />
             </intent-filter>
         </service>
-        <service android:name="org.chromium.chromecast.shell.CastWebContentsService"/>
+        <service android:name="org.chromium.chromecast.shell.CastWebContentsService"
+                 android:foregroundServiceType="specialUse">
+             <property android:name="android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE"
+                       android:value="cast" />
+        </service>
 
         <activity android:name="org.chromium.chromecast.shell.CastWebContentsActivity"
                   android:theme="@style/CastShellTheme"
diff --git a/third_party/android_sdk/README.chromium b/third_party/android_sdk/README.chromium
index d550ffccac8d5..deba9cba3b387 100644
--- a/third_party/android_sdk/README.chromium
+++ b/third_party/android_sdk/README.chromium
@@ -1,12 +1,12 @@
 Name: Android SDK
 URL: http://developer.android.com/sdk/index.html
-Version: 31
-  Android SDK Build-Tools 33.0.0
+Version: 34
+  Android SDK Build-Tools 34.0.0
   Android SDK Command-line Tools 7.0
   Android SDK Emulator 31.2.10
   Android SDK Patch Applier v4
-  Android SDK Platform-Tools 33.0.2
-  Android SDK Platform API 33
+  Android SDK Platform-Tools 34.0.3
+  Android SDK Platform API 34
   Android SDK Sources 31
   Google Cloud Messaging for Android Library, rev 3
 Security Critical: no
diff --git a/third_party/android_system_sdk/README.chromium b/third_party/android_system_sdk/README.chromium
index 63b429600cd0c..3772a6d5ba9fc 100644
--- a/third_party/android_system_sdk/README.chromium
+++ b/third_party/android_system_sdk/README.chromium
@@ -2,7 +2,7 @@ Name: Android System SDK
 URL: https://android.googlesource.com/platform/frameworks/base
 Short Name:  Android System SDK
 Version: 0
-Revision: TPB3.220513.017
+Revision: UPB3.230519.008
 License: GPL v2
 License File: LICENSE
 Security Critical: No
